pandas/core/reshape/merge.py

```diff
@@ -938,16 +938,23 @@ def _create_join_index(
         -------
         join_index
         """
-        if self.how in (how, "outer") and not isinstance(other_index, MultiIndex):
+        if self.how in (how, "outer") and not isinstance(index, MultiIndex):
             # if final index requires values in other_index but not target
             # index, indexer may hold missing (-1) values, causing Index.take
             # to take the final value in target index. So, we set the last
             # element to be the desired fill value. We do not use allow_fill
             # and fill_value because it throws a ValueError on integer indices
-            mask = indexer == -1
+            mask = other_indexer == -1
```

Comment:
Can you add a description for other_indexer to the docstring?

Reply:
Done. Did not see that it was missing previously.